tikv/kv.go

```diff
@@ -601,18 +614,36 @@ func (s *KVStore) updateSafeTS(ctx context.Context) {
 		}(ctx, wg, storeID, storeAddr)
 	}
 
-	txnScopeMap := make(map[string][]uint64)
+	for txnScope, storeIDs := range txnScopeMap {
+		s.updateMinSafeTS(txnScope, storeIDs)
+	}
+	wg.Wait()
+}
+
+// build txnScopeMap and judge whether it is needed to get safeTS from PD.
+// - if stores label are global, return get cluster min resolved ts from pd.
+// - if contains dc label store, return try to get it from TiKV.
+func (s *KVStore) buildTxnScopeMap(ctx context.Context, stores []*Store) (safeTS uint64, txnScopeMap map[string][]uint64, err error) {
+	isGlobal := true
```

Comment:
We do not need judge from the tikv labels, I think should judge from the config like the usage in tidb:
https://github.com/pingcap/tidb/blob/88225787f3c3da18323415c72208bce20876009a/expression/builtin_time.go#L6617-L6632

Reply:
go it